File: src/warning/ComplexMethodWarning.java
package warning;

import org.eclipse.jdt.core.dom.ASTNode;
import org.eclipse.jdt.core.dom.CompilationUnit;

public class ComplexMethodWarning extends Warning {
	private int cyclomaticComplexity;

	public ComplexMethodWarning(CompilationUnit unit, ASTNode node, String filename, int cyclomaticComplexity) {
		super(unit, node, filename);
		this.cyclomaticComplexity = cyclomaticComplexity;
	}

	@Override
	public String getMessage() {
		return "サイクロマチック数が大きい(" + cyclomaticComplexity + ") " + node.toString();
	}

}

File: src/warning/ComplexMethodWarning.kt
package warning

import org.eclipse.jdt.core.dom.ASTNode
import org.eclipse.jdt.core.dom.CompilationUnit
import util.Log
import warning.suggestion.*
import metricsplugin.views.metricstreeview.MetricsCategory
import codeanalyzer.ASTUtil

class ComplexMethodWarning(unit: CompilationUnit?, node: ASTNode?, filename: String?, cyclomaticComplexity: Int) : Warning(unit, node, filename) {
	private val cyclomaticComplexity: Int = cyclomaticComplexity
	
	override fun getMessage(): String {
		return "サイクロマチック数が大きい(" + cyclomaticComplexity  + ") " + ASTUtil.methodNameOf(node);
	}
	
	override fun suggestions(): List<Suggestion> {
		return arrayListOf(ExtractConditionsSuggestion(this), SplitMethodSuggestion(this), UsePolymorphismSuggestion(this))
	}
	
	override fun getParent(): MetricsCategory {
		return MetricsCategory.SIMPLE_METRICS;
	}
	
	override fun getPriority() = cyclomaticComplexity
}